pkg/cmd/browse/browse.go

```diff
@@ -123,7 +128,14 @@ func runBrowse(opts *BrowseOptions) error {
 	}
 
 	if opts.CommitFlag {
-		commit, err := git.LastCommit()
+		if opts.LastCommit == nil {
+			httpClient, err := opts.HttpClient()
+			if err != nil {
+				return err
+			}
+			opts.LastCommit = func() (*git.Commit, error) { return api.LastCommit(api.NewClientFromHTTP(httpClient), baseRepo) }
```

Comment:
OK, I see what you're doing here and it's a solid start, but I wonder if we can make this even more simpler & straightforward for the `runBrowse()` implementation. Right now, swapping out implementations of `LastCommit` like this is kind of awkward.

Ultimately, `runBrowse()` doesn't need to know about the type of lookup that will be performed. The only thing it wants to know is: what is the last commit for this repository? Of course, the definition of what "this repository" is depends on the `--repo` flag, but `runBrowse()` ideally should not be aware of whether `--repo` was used or not.

I propose to pass a concept of a "git client" to `runBrowse` via its opts struct. Something like:
```go
type BrowseOptions struct {
  GitClient gitClient
}

type gitClient interface {
  LastCommit() (*git.Commit, error)
}
```

What implements the `gitClient` interface? Right now there are no implementations, but you could create some: 1. The first and obvious implementation is the one that consults `git.LastCommit()` and therefore reads from the local git repository of the current directory;
2. The second implementation is the one that uses `api.LastCommit(httpClient, repo)` to fetch commit information from the API.

Either of these implementations could be assigned to `opts.GitClient` by the command's RunE block based on whether or not the `--repo` flag has been passed.